app/modules/loan/endpoints_loan.py

```diff
@@ -872,18 +872,7 @@ async def return_loan(
         returned=True,
         returned_date=datetime.now(UTC),
     )
-
-    message = Message(
-        context=f"loan-new-{loan.id}-end-notif",
-        is_visible=False,
-        title="",
-        content="",
-        expire_on=datetime.now(UTC) + timedelta(days=3),
-    )
-    await notification_tool.send_notification_to_user(
-        user_id=loan.borrower_id,
-        message=message,
-    )
+    await scheduler.cancel_job(f"loan_end_{loan.id}")
```

Comment:
As the job was created by the `notification_manager` which mask the usage of the scheduler, you could create a `notification_manager.cancel_notification(notification_id: str)`

Reply:
I think the method that send a future notification could always cancel existing notifications with the same id. I don't think there will be a case where we want to send two notifications with the same id